### PR TITLE
Adding deleted_at check on variant_stock move

### LINKED ISSUE #5335: After deleting a product I can't cancel an order containing that product
## Description

Once a product has been soft-deleted it still appears in orders placed.

If I then want to cancel that order I cannot.

message = 'You need to save the variant to create a stock item before you can set stock levels.'

https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/5da4ec3ec076fc0019b57ad6?filters[event.since][0]=30d&filters[error.status][0]=open

## Expected Behavior

I want to be able to cancel orders that contain deleted products

## Actual Behaviour

I cannot cancel an order than contains a deleted product

## Steps to Reproduce

1. Create an order with a product
2. Delete the product
3. Cancel the order
4. Death by slug

## Workaround

Possibly by recreating the product and then cancelling the order? I haven't tested this.

## Severity

bug-s3: a feature is broken but there is a workaround of adding a negative adjustment.
I'm not sure if there is a workaround. This seems to mostly impact people making test orders on the system. This doesn't seem to impact people so often once they start trading.

## Your Environment

* Version used: UK 2.9.6
* Browser name and version:
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by georgethoppil**
#### What? Why?

Closes #5335 

If the variant has been deleted, it will not attempt to restock stock items 

#### What should we test?
1. Create an order with a product
2. Delete the product
3. Cancel the order
4. The order should be cancelled and not throw an error

### CODE REVIEW

**Review by mkllnk — APPROVED** [review]
Beautiful solution, thank you. :bouquet:

File: spec/models/spree/order_spec.rb [hunk 1]
```diff
@@ -283,6 +283,13 @@
       order.completed_at = Time.zone.now
       expect(order.can_cancel?).to be_truthy
     end
+
+    it "should cancel order if product is soft deleted" do
+      o = FactoryBot.create(:completed_order_with_totals)
+      o.line_items.first.variant.product.tap(&:destroy)
+      o.cancel!
+      expect(Spree::Order.by_state(:canceled)).to include o
+    end
```
Comment (Matt-Yorkley) [inline spec/models/spree/order_spec.rb:292]: Nice! I've just got some syntax suggestions (let/create syntax and avoiding single-letter-variables):

```suggestion
    context "with a deleted product" do
      let(:order) { create(:completed_order_with_totals) }
      
      it "should cancel order without error" do
        order.line_items.first.variant.product.tap(&:destroy)
        order.cancel!
        expect(Spree::Order.by_state(:canceled)).to include order
      end
    end  
```

Comment (georgethoppil) [reply spec/models/spree/order_spec.rb]: great suggestion, thanks

**Review by Matt-Yorkley — APPROVED** [review]
LGTM :ok_hand:

**Comment by filipefurtad0** [comment]
Hey @Dmoment ,

This looks good, the video below displays an order with a deleted product. Cancelling it does not trigger the error 500 :tada: 

Peek 2022-01-26 17-30

Thank you.
Merging!

**Pull request merged**